File: app/src/main/java/loshica/quiz/NameDialog.java
package loshica.quiz;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class NameDialog extends DialogFragment {

    private EditText et;
    NameDialogListener listener;

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(requireActivity());
        LayoutInflater inflater = requireActivity().getLayoutInflater();
        @SuppressLint("InflateParams") View root = inflater.inflate(R.layout.dialog_name, null);
        et = root.findViewById(R.id.name_dialog_input);

        return builder
            .setView(root)
            .setTitle(R.string.name_dialog_title)
            .setPositiveButton(R.string.ok, (dialog, which) -> listener.name(et.getText().toString()))
            .create();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        try {
            listener = (NameDialogListener) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(context.toString() + e);
        }
    }

    public interface NameDialogListener {
        void name(String username);
    }
}

File: app/src/main/java/loshica/quiz/view/NameDialog.kt
package loshica.quiz.view

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import loshica.quiz.R
import loshica.quiz.databinding.DialogNameBinding
import loshica.vendor.view.LOSDialogBuilder

class NameDialog : DialogFragment() {

    private var _b: DialogNameBinding? = null
    private val b get() = _b!!
    private var listener: NameDialogListener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = LOSDialogBuilder(requireActivity())
        _b = DialogNameBinding.inflate(requireActivity().layoutInflater)

        return builder
            .setView(b.root)
            .setTitle(R.string.name_dialog_title)
            .setPositiveButton(R.string.ok) { dialog, _ ->
                if (b.nameInput.text.toString() != "") listener!!.name(b.nameInput.text.toString())
                else dialog.cancel()
            }
            .setNegativeButton(R.string.cancel, null)
            .create()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _b = null
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        listener = try { context as NameDialogListener }
        catch (e: ClassCastException) { throw ClassCastException(context.toString() + e) }
    }

    interface NameDialogListener {
        fun name(playerName: String)
    }
}